lib/src/realm_class.dart

```diff
@@ -483,6 +483,40 @@ class Realm implements Finalizable {
       throw RealmError('Starting a write transaction on a frozen Realm is not allowed.');
     }
   }
+
+  /// Compacts a Realm file. A Realm file usually contains free/unused space.
+  ///
+  /// This method removes this free space and the file size is thereby reduced.
+  /// Objects within the Realm file are untouched.
+  /// Note: The file system should have free space for at least a copy of the Realm file. This method must not be called inside a transaction.
+  /// The Realm file is left untouched if any file operation fails.
+  static Future<bool> compact(Configuration config) {
```

Comment:
Why is this a future?

Reply:
This is cause this operation is long running and also we may decide to to that on a background isolate later. This is why I changed the method signature to accept a Configuration object.

Reply:
I don't understand what you mean. How a Future result is a non blocking operation. If you await on it when you invoke the method you are blocking at the point of the call and waiting for the result to continue from that point forward.

Reply:
In other words a method that returns a Future does not guarantee it will not execute synchronously

Reply:
we will continue the discussion offline

Reply:
I think people are right to expect that the thread will not be blocked for more than a few ms when awaiting an async function. Even if no such guarantees exists, it is still the expectation. 

Perhaps we can wait with exposing this until we figure out how to move the work onto a background isolate?

Reply:
fixed